### PR TITLE
jit: enable build on Windows 

### PR DESCRIPTION
**Pull request by mathetake**
part of #154. Currently we only implement the mmap for `linux` and `darwin`,
therefore JIT is only available for these GOOSes. This commit at least makes it 
possible for Windows to do `go build ./...`.

### CODE REVIEW

File: wasm/jit/mmap_darwin.go [hunk 1]
```diff
@@ -1,5 +1,24 @@
+//go:build darwin
+// +build darwin
+
 package jit
 
 import "syscall"
 
 const mmapFlags = syscall.MAP_ANON
+
+// Copy the code into the executable region
```
Comment (codefromthecrypt) [inline wasm/jit/mmap_darwin.go:10]:
```suggestion
// mmapCodeSegment copies the code into the executable region
```

File: wasm/jit/mmap_darwin.go [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+// and returns the byte slice of the region.
+func mmapCodeSegment(code []byte) ([]byte, error) {
+	mmapFunc, err := syscall.Mmap(
+		-1,
+		0,
+		len(code),
+		syscall.PROT_READ|syscall.PROT_WRITE|syscall.PROT_EXEC, syscall.MAP_PRIVATE|mmapFlags,
```
Comment (codefromthecrypt) [inline wasm/jit/mmap_darwin.go:17]: comment why these flags using a link if necessary

File: wasm/jit/mmap_linux.go [hunk 3]
```diff
@@ -1,5 +1,24 @@
+//go:build linux
+// +build linux
+
 package jit
 
 import "syscall"
 
 const mmapFlags = syscall.MAP_ANONYMOUS
+
+// Copy the code into the executable region
```
Comment (codefromthecrypt) [inline wasm/jit/mmap_linux.go:10]:
```suggestion
// mmapCodeSegment copies the code into the executable region
```

File: wasm/jit/mmap_other.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
+//go:build !darwin && !linux
+// +build !darwin,!linux
+
+package jit
+
+const mmapFlags = 0
+
+// Copy the code into the executable region
```
Comment (codefromthecrypt) [inline wasm/jit/mmap_other.go]: change this to `mmapCodeSegment is unsupported on windows.`

**Review by codefromthecrypt — COMMENTED** [review]
instead of copy/paste, maybe resurrect `mmap.go` and include `mmapCodeSegment` as `mmapCodeSegmentPosix` or some better name. then instead of 2 files for darwin and linux, use one with flags for both and in that do..

```
func mmapCodeSegment().. {
   return mmapCodeSegmentPosix()
}
```

that way mmap.go can have unit tests and also those tests don't need to be copy/pasted

wdyt?

File: wasm/jit/mmap_other.go [hunk 5]
```diff
@@ -0,0 +1,8 @@
+//go:build !darwin && !linux
```
Comment (codefromthecrypt) [inline wasm/jit/mmap_other.go:1]: ps I think it is fine to rename this to _windows and wait for someone to ask for something besides darwin and linux. we would likely want CI if that were the case anyway

**Review by codefromthecrypt — APPROVED** [review]

File: wasm/jit/mmap_darwin.go (hunk 2 above)
Comment (codefromthecrypt) [reply wasm/jit/mmap_darwin.go]: @mathetake ready except I don't know what to put here

Comment (mathetake) [reply wasm/jit/mmap_darwin.go]: https://github.com/tetratelabs/wazero/pull/155/commits/2f3a70f403e9ea78dbb66a4ea239203d54871f7d Added comments and link!

**Comment by mathetake** [comment]
Thanks for the look and polish!

**Comment by codefromthecrypt** [comment]
thanks @mathetake your comments taught me some things!

**Pull request merged**